core/embed/rust/src/ui/model_tt/component/page.rs

```diff
@@ -179,18 +245,60 @@ impl Component for ScrollBar {
             self.area.center().x,
             self.area.center().y - (count / 2) * interval,
         );
-        for i in 0..self.page_count {
-            display::rounded_rect(
-                Rect::from_center_and_size(dot, Self::DOT_SIZE),
-                if i == self.active_page {
-                    theme::FG
-                } else {
-                    theme::GREY_LIGHT
-                },
+        if self.has_previous_page() {
+            display::icon(
+                dot - Offset::new(0, Self::ARROW_SPACE),
+                Self::ICON_UP,
+                theme::FG,
                 theme::BG,
-                theme::RADIUS,
             );
+        }
+        for i in 0..self.page_count {
+            let icon = if i == self.active_page {
+                Self::ICON_ACTIVE
+            } else {
+                Self::ICON_INACTIVE
+            };
+            display::icon(dot, icon, theme::FG, theme::BG);
             dot.y += interval;
         }
+        if self.has_next_page() {
+            display::icon(
+                dot + Offset::new(0, Self::ARROW_SPACE - interval),
+                Self::ICON_DOWN,
+                theme::FG,
+                theme::BG,
+            );
+        }
+    }
+}
+
+pub struct PageLayout {
+    pub content_single_page: Rect,
+    pub content: Rect,
+    pub scrollbar: Rect,
+    pub buttons: Rect,
+}
+
+impl PageLayout {
+    const BUTTON_SPACE: i32 = 6;
+    const SCROLLBAR_WIDTH: i32 = 10;
+    const SCROLLBAR_SPACE: i32 = 10;
+
+    pub fn new(area: Rect) -> Self {
+        let (content, buttons) = area.hsplit(-Button::<&str>::HEIGHT);
```

Comment:
I'd personally opt for `hor_split` and `vert_split`, I sometimes get lost in the `v`/`h` in these calculations. Or, even better, left/right + top/bottom instead of positive/negative values,  because we usually want one or the other. Just a personal preference, feel free to ignore :)

Reply:
Yeah, I've also had to pause and think about which is which. Should be fixed in e0abc6c, please take a look.